source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -869,7 +869,7 @@ export class TextBlock extends BaseTextBlock {
 
                 if (hostConfig.supportsInteractivity) {
                     element.tabIndex = 0
-                    element.setAttribute("role", "button");
+                    element.setAttribute("role", Utils.roleFromAction(this.selectAction));
```

Comment:
This should be a `getAriaRole()` method off the `Action` class, that returns "button" by default. It would be overridden in `OpenUrlAction` to return "link". With the way you have it now, it's not possible for a custom action type to decide what aria role it's associated with.